OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -244,8 +271,10 @@ protected virtual void Tick(Actor self)
 				FinishProduction();
 			}
 
-			if (queue.Count > 0)
-				queue[0].Tick(playerResources);
+			if (queue.Count < 1 || allProductionPaused)
+				return;
```

Comment:
Changing this to use an early return feels unnecessary.  Can you please stick with `queue.Count > 0 && !allProductionPaused`?